python/rikai/types/geometry.py

```diff
@@ -440,3 +448,207 @@ def to_dict(self) -> dict:
             "height": self.height,
             "heading": self.heading,
         }
+
+
+class Mask(ToNumpy, ToDict):
+    """2-d Mask over an image
+
+    This 2D mask can be built from:
+
+    - A binary-valued (0 or 1) 2D-numpy matrix (:py:class:`Mask.Type.MASK`).
+    - A Run Length Encoded (RLE) data. It supports both row-based RLE
+      (:py:class:`Mask.Type.RLE`) or column-based RLE
+      (:py:class:`Mask.Type.COCO_RLE`) which is used in the Coco dataset.
+    - A Polygon ``[x0, y0, x1, y1, ..., xn, yn]``
+
+    Parameters
+    ----------
+    data: list or :py:class:`np.ndarray`
+        The mask data. Can be a numpy array or a list.
+    height: int, optional
+        The height of the image this mask applies to.
+    width: int, optional
+        The width of the image this mask applies to.
+    mask_type: :py:class:`Mask.Type`
+        The type of the mask.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        from pycocotools.coco import COCO
+        from rikai.types import Mask
+
+        coco = COCO("instance_train2017.json")
+        ann = coco.loadAnns(ann_id)
+        image = coco.loadImgs(ann["image_id"])
+        if ann["iscrowed"] == 0:
+            mask = Mask.from_polygon(
+                ann["segmentation"],
+                height=image["height"],
+                width=image["width],
+            )
+        else:
+            mask = Mask.from_coco_rle(
+                ann["segmentation"]["counts"],
+                height=image["height"],
+                width=image["width],
+            )
+
+    """
+
+    __UDT__ = MaskType()
+
+    class Type(Enum):
+        """Mask type."""
+
+        MASK = 0
+        POLYGON = 1
+        RLE = 2
+        COCO_RLE = 3  # COCO style RLE, Column-based
+
+    def __init__(
+        self,
+        data: Union[list, np.ndarray],
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+        mask_type: Mask.Type = Type.POLYGON,
+    ):
+        self.type = mask_type
+        self.data = data
+        if self.type == Mask.Type.MASK:
+            self.height = data.shape[0]
+            self.width = data.shape[1]
+        else:
+            assert (
```

Comment:
nit: asserts get removed if optimized mode is turned on. So this is likely to result in surprising behavior based on user configurations

Reply:
ok, changed this to `raise ValueError`